awsbestpractices/s3-with-prefix/S3WithPrefix.java

```diff
@@ -0,0 +1,69 @@
+package example;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.model.*;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class Example {
+
+    private final AmazonS3 s3Client;
+
+    public static double conditionWithPagination1(List<String> inputURIs) {
+        Long numBytes = 0L;
+        for (String uri : inputURIs) {
+            AmazonS3URI s3URI = new AmazonS3URI(uri);
+            ObjectListing objects = null;
+            do {
+                if (objects == null) {
+                    objects = s3Client.listObjects(s3URI.getBucket(), s3URI.getKey());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).